plasmapy/__init__.py

```diff
@@ -33,12 +33,36 @@
  __citation__      --- PlasmaPy citation instructions
 
 """
+__all__ = [
+    'addons',
+    'classes',
+    'data',
+    'diagnostics',
+    'formulary',
+    'particles',
+    'simulation',
+    'utils',
+    'online_help',
+]
+
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
 # Packages may add whatever they like to this file, but
 # should keep this content at the top.
 # ----------------------------------------------------------------------------
 import pkg_resources
+import sys
+
+from . import (
+    addons,
+    classes,
+    data,
+    diagnostics,
+    formulary,
+    particles,
+    simulation,
+    utils,
+)
```

Comment:
IIRC (and this really should have been documented in here...) this automatically-import-every-subpackage style of doing things (as opposed to, e.g., the scipy way) was done in order to ensure dependencies for some subpackages are optional if you're not willing to use them. We've been better about moving optional deps into functionality code recently, though, and I've been meaning to crack down on the optional ones anyway (because matplotlib is not really difficult to install when you've seen VTK!), so I'm all for this anyway.

Reply:
This will automatically import everything.  However, I'm doing this to control the namespace of `plasmapy`.  I think properly exposing the namespace makes the package much more discoverable for new and old users.  Someone can actually find stuff using `dir()`.  Before this `dir(plasmapy)` would show a bunch of dunder objects, now it'll show all the available sub-packages.

This does lead to longer package loading times, but I'm okay with this for the increased user friendliness.  To counter act these loading times (especially as the package gets larger), I say we encourage specific module imports in our documentation (e.g. `import plasmapy.addons as pads` or `import plasmapy.formulary as pform`).